task/buildah-oci-ta/0.2/buildah-oci-ta.yaml

```diff
@@ -288,14 +288,12 @@ spec:
 
         BUILDAH_ARGS=()
 
-        BASE_IMAGES=$(grep -i '^\s*FROM' "$dockerfile_path" | sed 's/--platform=\S*//' | awk '{print $2}' | (grep -v ^oci-archive: || true))
+        BASE_IMAGES=$(dockerfile-json "$dockerfile_path" | jq -r '.Stages[] | select(.From | .Stage or .Scratch | not) | .BaseName')
```

Comment:
It is now possible to have an empty value here, right whereas before there was always going to be at least one value?

Will this break any functionality elsewhere?

Reply:
Yes, previously there was always at least one value - e.g when it was FROM scratch, but now we are omitting that. That's why I could remove all those if conditions that are checking for "scratch". 

I tested it with a couple of builds and the e2e tests are passing. And from the logic of the buildah it does not seem anything should break, since when we are passing that further (for example to the sbom), the "scratch" was omitted anyway.

Reply:
An empty Dockerfile should seldom happen. However, if it happens accidentally, dockerfile-json will report an error like `[dockerfile-json 1.0.8] error: parse "./Dockerfile": dockerfile/parser.Parse file with no instructions`. And `set -o pipefail` is not set in the `build` step.

Reply:
It is good that the e2e tests are passing but I don't have much faith in the robustness of the e2e tests lately (i.e. do they test an image which is just `FROM scratch`)? Would you be able to test this change in a pipeline that has a simple `FROM scratch` Containerfile?

If you need a file, you can customize this pipeline: https://github.com/konflux-ci/olm-operator-konflux-sample/blob/main/.tekton/gatekeeper-operator-bundle-pull-request.yaml#L35 and just modify the Dockerfile to remove the first stage (or better yet, test it with the first stage and without).

Reply:
We do have a FROM scratch test, see https://github.com/konflux-ci/e2e-tests/commit/9c5c74c7b8c1f23b70b2f48aabe86a78ab365425 and https://github.com/konflux-ci/build-definitions/commit/5573eb35e44f486f539152d184f717bedb3b5542. But I tested it anyway, once again, now also with multiple stages as has the dockerfile you linked.